src/js/frontend/src/components/login/LoginForm.tsx

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import { Field, Form, Formik } from "formik";
+import { GoogleSSO } from "./GoogleSSO";
+
+import { loginService } from "../../services/login/loginService";
+import { yupValidationSchema } from "./yupValidationSchema";
+
+export interface FormValues {
+  username: string;
+  password: string;
+}
+
+export interface FormProps {
+  onSubmit: (formValue: FormValues) => void;
+}
+
+const LoginForm = ({ onSubmit }: FormProps) => {
+  const [state, setState] = React.useState({
+    loginStatus: true, // Boolean represented as true when user is successfully authenticated,
+    // false when there's an auth error, token has been removed or user has been logged out
+  });
+
+  const handleSubmit = async (values: FormValues) => {
+    const successfulLoginResponse = await loginService(values.username, values.password);
+
+    if (successfulLoginResponse) {
+      window.history.replaceState("#/login", "", "#/");
+      window.location.reload();
+    } else {
+      setState({
+        ...state,
+        loginStatus: false,
+      });
+    }
+    onSubmit(values);
+  };
+
+  return (
+    <div data-testid="googleSSOContainer">
+      <h1 data-testid={"Grapl"}> Grapl </h1>
+
+      <img src={"./assets/graplLogoFull.svg"} alt={"Grapl Logo"} />
+
+      <GoogleSSO state={state} setState={setState} />
+
+      <Formik
+        data-testid="form"
+        initialValues={{ username: "", password: "" }}
+        validationSchema={yupValidationSchema}
+        onSubmit={handleSubmit}
+      >
+        {({ errors, touched }) => (
+          <Form>
+            <Field data-testid={"username"} name="username" type="text" placeholder="Username" />
+            {touched.username && errors.username && <div>{errors.username}</div>}
+            <Field
+              data-testid={"password"}
+              name="password"
+              type="password"
+              placeholder="Password"
+            />{" "}
+            <br />
+            {touched.password && errors.password && <div>{errors.password}</div>}
+            <button data-testid={"button"} name="submitButton" type="submit">
+              SUBMIT
+            </button>
+            {state.loginStatus && <div data-testid={"loginError"}>Unsuccessful Login</div>}
```

Comment:
Won't this display "Unsuccessful Login" only when the user successfully logged in?

Reply:
Resolved